packages/react-jss/tests/types/createUseStyles.ts

```diff
@@ -172,53 +171,14 @@ const noThemeArg4ClassesPass2 = noThemeArg4({...expectedCustomProps, theme: expe
 const noThemeArg5 = createUseStyles<string, MyProps, MyTheme>({
   singleNest: {
     fontWeight: 'bold',
-    singleValue: ({property, theme}) => '',
-    nestOne: ({property, theme}) => ({
+    // @ts-expect-error
+    nestOne: ({property, theme}: MyProps & {theme: MyTheme}) => ({
       color: 'red',
-      // @ts-expect-error
-      nothingAllowed: ({theme: innerTheme, ...innerProps}) => ''
+      display: () => 'block'
     })
   }
 })
 
-// Nested declarations are banned (double nest test)
-const noThemeArg6 = createUseStyles<string, MyProps, MyTheme>({
```

Comment:
I don't clearly understand what this is testing, because nesting is not generally disallowed, its just only working with  nested syntax with &